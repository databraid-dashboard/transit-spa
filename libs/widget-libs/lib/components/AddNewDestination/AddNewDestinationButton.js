var _jsxFileName = '../../src/components/AddNewDestination/AddNewDestinationButton.jsx',
    _this = this;

import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'semantic-ui-react';

var AddNewDestinationButton = function AddNewDestinationButton(_ref) {
  var onClick = _ref.onClick;
  return React.createElement(
    'div',
    { id: 'button-div', __source: {
        fileName: _jsxFileName,
        lineNumber: 6
      },
      __self: _this
    },
    React.createElement(Button, {
      className: 'transit-button',
      circular: true,
      icon: 'plus',
      size: 'massive',
      color: 'blue',
      onClick: onClick,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 7
      },
      __self: _this
    })
  );
};

AddNewDestinationButton.propTypes = {
  onClick: PropTypes.func.isRequired
};

AddNewDestinationButton.defaultProps = {
  onClick: function onClick() {}
};

export default AddNewDestinationButton;