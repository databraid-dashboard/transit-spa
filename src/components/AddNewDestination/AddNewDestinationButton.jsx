import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'semantic-ui-react';
import './button.css';

const AddNewDestinationButton = ({ onClick }) =>
  (<div id="button-div">
    <Button circular icon="plus" size="huge" color="blue" onClick={onClick} />
  </div>);

AddNewDestinationButton.propTypes = {
  onClick: PropTypes.func.isRequired,
};

AddNewDestinationButton.defaultProps = {
  onClick: () => {},
};

export default AddNewDestinationButton;
