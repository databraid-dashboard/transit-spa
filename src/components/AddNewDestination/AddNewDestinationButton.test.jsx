import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddNewDestinationButton from './AddNewDestinationButton';

describe('AddNewDestinationButton', () => {
  it('should render button', () => {
    const component = shallow(<AddNewDestinationButton />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
