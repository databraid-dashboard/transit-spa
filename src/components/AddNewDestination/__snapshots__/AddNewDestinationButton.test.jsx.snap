// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNewDestinationButton should render button 1`] = `
<div
  id="button-div"
  style={
    Object {
      "marginLeft": "0.5rem",
    }
  }
>
  <Button
    as="button"
    circular={true}
    className="transit-button"
    color="blue"
    icon="plus"
    onClick={[Function]}
    size="large"
  />
</div>
`;
